app/controllers/attachment_references_controller.rb

```diff
@@ -0,0 +1,14 @@
+class AttachmentReferencesController < ApplicationController
+  load_resource :attachment_reference
+  before_action :authorize_resource
+
+  def show
+    redirect_to @attachment_reference.url
+  end
+
+  private
+
+  def authorize_resource
+    authorize!(:read, @attachment_reference.attachable)
```

Comment:
OK, so we need to authorize both the attachable when the page is loaded, and also the resource. This is in case the user copies the URL from the forum post, we want to make sure that the attachment's access is permitted.